### PR TITLE
samples: sysbuild: updated IPC RPMsg service to use CPU cluster SoC info

### PR DESCRIPTION
**Pull request by tejlmand**
Move the Kconfig symbols SOC_ESP32_PROCPU, SOC_ESP32_APPCPU,
SOC_ESP32S3_PROCPU, and SOC_ESP32S3_APPCPU.

The CPU cluster is defined in espessifc/soc.yml and should therefore
be available in the HWMv2 Kconfig.soc tree.

This will allow sysbuild to test for the CPU cluster when targeting
remote board for a build.

Update espressif boards accordingly.

### CODE REVIEW

**Comment by tejlmand** [comment]
@sylvioalves  I have done first move of the PROCPU / APPCPU for the RPMsg sample use-case in commit https://github.com/zephyrproject-rtos/zephyr/pull/69749/commits/74b33eabc620e65066d61596b895e2d1afd86737.

Please followup for other esp32 / esp32s3 based boards, and fell free to force-push into this PR, or push `fixup` commits, then I'll rebase afterwards.

**Comment by sylvioalves** [comment]
> @sylvioalves I have done first move of the PROCPU / APPCPU for the RPMsg sample use-case in commit [74b33ea](https://github.com/zephyrproject-rtos/zephyr/commit/74b33eabc620e65066d61596b895e2d1afd86737).
> 
> Please followup for other esp32 / esp32s3 based boards, and fell free to force-push into this PR, or push `fixup` commits, then I'll rebase afterwards.

@tejlmand, thanks for bringing this. I would start handling it this week.

File: boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom [hunk 1]
```diff
@@ -3,3 +3,5 @@
 
 config BOARD_ESP32_DEVKITC_WROOM
 	select SOC_ESP32_WROOM_32UE_N4
+	select SOC_ESP32_PROCPU if BOARD_ESP32_DEVKITC_WROOM_ESP32_PROCPU
+	select SOC_ESP32_APPCPU if BOARD_ESP32_DEVKITC_WROOM_ESP32_APPCPU
```
Comment (sylvioalves) [inline boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom:7]: @nordicjm moving this SOC definition into here was my initial approach. However, you suggested me to move this into `board/Kconfig` instead. Any comments?

Comment (sylvioalves) [reply boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom:7]: @tejlmand Anyway, this move won't work. Shall I restore to `kconfig` instead? ( I mean, keep in there):
```c
error: SOC_ESP32_PROCPU (defined at soc/espressif/esp32/Kconfig:20) is assigned in a configuration
file, but is not directly user-configurable (has no prompt). It gets its value indirectly from other
symbols.
```

Comment (nordicjm) [reply boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom:7]: If you are selecting an actual core then yes it belongs here, but that was not the case in the original PR, these selected something relating to an AMP configuration

File: share/sysbuild/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2023 Nordic Semiconductor
+# Copyright (c) 2021-2023 Nordic Semiconductor
```
Comment (nordicjm) [inline share/sysbuild/CMakeLists.txt:1]: This commit has been merged already

**Review by nordicjm — APPROVED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
(Ack for changes, nack until first commit is removed)

**Comment by tejlmand** [comment]
> (Ack for changes, nack until first commit is removed)

No need to nack, we're still waiting @sylvioalves to push the changes to update remaining nxp boards.
Hence the `DNM` label.

@sylvioalves feel free to rebase the branch, so that both of us don't force-push the branch.

**Comment by sylvioalves** [comment]
> > (Ack for changes, nack until first commit is removed)
> 
> No need to nack, we're still waiting @sylvioalves to push the changes to update remaining nxp boards. Hence the `DNM` label.
> 
> @sylvioalves feel free to rebase the branch, so that both of us don't force-push the branch.

Sure, working on that right now. Will push soon.

File: boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom (hunk 1 above)
Comment (sylvioalves) [reply boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom:7]: @tejlmand, `SOC_ESP32_PROCPU` does not select a SoC. It only tells the build system which core is the build intended for, thus allowing proper SoC initialization (one or the other). 
1. Should we rename this and remove the `SOC_` prefix?
2. If we keep, should this be kept where it is instead moving to `Kconfig.soc`?

Comment (tejlmand) [reply boards/espressif/esp32_devkitc_wroom/Kconfig.esp32_devkitc_wroom:7]: it seems like two symbols might be needed.
I'm not into the details on how the current `SOC_ESP32_PROCPU` / `SOC_ESP32_APPCPU`  symbols are used, but this thread indicates that it's different than a CPU cluster symbol reflecting the `soc.yml` setting.

I'll leave it to you to decide if the symbols can be united to a single one, or if two independent symbols are required.

But we should have a symbol which can be selected by the board to indicate the CPU Cluster for which the build is targeted. Especially for use-case like the ones in this PR.

**Comment by tejlmand** [comment]
> Sure, working on that right now. Will push soon.

@sylvioalves any updates on this ?

**Review by nordicjm — APPROVED** [review]

**Review by tejlmand — COMMENTED** [review]
lgtm, @sylvioalves you may give this a +1 on my behalf.

Thanks for taking this to the finish line.

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**